proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -14074,37 +14032,37 @@ lemma send_signal_WaitingNtfn_helper:
          apply (fastforce dest: st_tcb_at_idle_thread)
         defer
         apply (clarsimp simp: not_cur_thread_def ct_in_state_def tcb_at_kh_simps pred_map_def)
-      subgoal for s
-      apply (subgoal_tac "valid_ntfn ntfn s")
-       apply (clarsimp simp: valid_ntfn_def split: list.splits option.splits)
-       apply (case_tac wnlist; simp)
-      by (frule invs_valid_objs, fastforce simp: valid_objs_def dom_def obj_at_def valid_obj_def)
+    subgoal for s
+    apply (subgoal_tac "valid_ntfn ntfn s")
+     apply (clarsimp simp: valid_ntfn_def split: list.splits option.splits)
+     apply (case_tac wnlist; simp)
+    by (frule invs_valid_objs, fastforce simp: valid_objs_def dom_def obj_at_def valid_obj_def)
      apply (rule valid_sched_not_runnable_scheduler_act_not; clarsimp simp: valid_sched_def tcb_at_kh_simps pred_map_def)
     apply (rule valid_sched_not_runnable_not_in_release_q; clarsimp simp: valid_sched_def tcb_at_kh_simps pred_map_def)
     subgoal for s
-      apply (prop_tac "ipc_queued_thread (hd wnlist) s", simp add: tcb_at_kh_simps pred_map_def)
-      apply (rule released_ipc_queues_blocked_on_recv_ntfn_D, simp)
-      by (fastforce elim: pred_map_weakenE simp: tcb_at_kh_simps is_blocked_thread_state_defs)
-    (* sym_refs *) (* FIXME RT: this is nasty, can it be improved? *)
-    apply (case_tac wnlist; clarsimp)
-    apply (rule delta_sym_refs[OF invs_sym_refs], assumption)
-     apply (clarsimp split: if_splits)
-     apply (intro conjI; intro impI;
-            fastforce simp: obj_at_def state_refs_of_def get_refs_def2 ntfn_q_refs_of_def
-                     dest!: symreftype_inverse'
-                     split: option.splits list.splits)
-    apply (frule invs_valid_objs)
-    apply (drule (1) valid_objs_ko_at, clarsimp simp: valid_obj_def valid_ntfn_def)
-    apply (clarsimp split: if_splits simp: pred_tcb_at_def obj_at_def)
-     apply (intro conjI impI;
-            clarsimp simp: tcb_st_refs_of_def state_refs_of_def
-                     split: option.splits if_splits list.splits thread_state.splits
-                     dest!: symreftype_inverse')
-              apply (clarsimp simp: get_refs_def2)+
-    by (intro conjI impI;
-        fastforce simp: is_tcb obj_at_def state_refs_of_def get_refs_def2 ntfn_q_refs_of_def
-                  dest!: symreftype_inverse'
-                  split: option.splits list.splits if_splits)
+    apply (prop_tac "ipc_queued_thread (hd wnlist) s", simp add: tcb_at_kh_simps pred_map_def)
+    apply (rule released_ipc_queues_blocked_on_recv_ntfn_D, simp)
+    by (fastforce elim: pred_map_weakenE simp: tcb_at_kh_simps is_blocked_thread_state_defs)
+      (* sym_refs *) (* FIXME RT: this is nasty, can it be improved? *)
```

Comment:
I suppose this is more for the `refill_unblock_check` everywhere PR, but I'm fairly sure I was able to improve, or maybe even get rid of, all these `sym_refs` bits in the draft I had.

Reply:
that is not my comment, that’s an existing one. I changed the indentation only. You can see the diff without whitespace changes by changing the GitHub setting, I think.